src/main/java/hudson/plugins/gearman/MyGearmanWorkerImpl.java

```diff
@@ -634,7 +634,8 @@ private void submitFunction(GearmanFunction fun) {
             if (executorService == null) {
```

Comment:
This check should be removed (and similarly in the rest of the code) in favor of tagging this constructor argument as  @NonNull.

Reply:
See my other comment, it is actually a feature flag which would probably have better be done using `false` rather than `null`.  Turns out there is never an executorService anywhere and it is always null.

I had suspicions when seeing another part of the code calling `executor.shutDown()` and `executorService.shutdownNow()` without waiting for jobs to terminate. It is wrapped under a conditional `executorService != null` which is always false and the shutdown code is never invoked.

Reply:
So sounds like this should stay in place until the executor is implemented properly, or completely removed.